src/mlpack/bindings/R/print_R.cpp

```diff
@@ -0,0 +1,147 @@
+/**
+ * @file bindings/R/print_R.cpp
+ * @author Yashwant Singh Parihar
+ *
+ * Implementation of utility PrintR() function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include "print_R.hpp"
+#include "strip_type.hpp"
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/util/hyphenate_string.hpp>
+
+using namespace mlpack;
+using namespace std;
+
+namespace mlpack {
+namespace bindings {
+namespace r {
+
+
+/**
+ * Print the code for a .R binding for an mlpack program to stdout.
+ */
+void PrintR(const util::ProgramDoc& programInfo,
+             const string& functionName)
+{
+  // Restore parameters.
+  CLI::RestoreSettings(programInfo.programName);
+
+  const map<string, util::ParamData>& parameters = CLI::Parameters();
+  typedef map<string, util::ParamData>::const_iterator ParamIter;
+
+  // First, let's get a list of input and output options.  We'll take two passes
+  // so that the required input options are the first in the list.
+  vector<string> inputOptions, outputOptions;
+  for (ParamIter it = parameters.begin(); it != parameters.end(); ++it)
+  {
+    const util::ParamData& d = it->second;
+    if (d.input && d.required)
+    {
+      // Ignore some parameters.
+      if (d.name != "help" && d.name != "info" &&
+          d.name != "version")
+        inputOptions.push_back(it->first);
+    }
+    else if (!d.input)
+    {
+      outputOptions.push_back(it->first);
+    }
+  }
+
+  for (ParamIter it = parameters.begin(); it != parameters.end(); ++it)
+  {
+    const util::ParamData& d = it->second;
+    if (d.input && !d.required &&
+        d.name != "help" && d.name != "info" &&
+        d.name != "version")
+      inputOptions.push_back(it->first);
+  }
+
+  MLPACK_COUT_STREAM << "#' @export" << endl;
+  MLPACK_COUT_STREAM << functionName << " <- function(";
+  size_t indent = functionName.size() + 13 /* <- function(*/;
+  for (size_t i = 0; i < inputOptions.size(); ++i)
+  {
+    const util::ParamData& d = parameters.at(inputOptions[i]);
+
+    if (i != 0)
+      MLPACK_COUT_STREAM << "," << endl << std::string(indent, ' ');
+
+    CLI::GetSingleton().functionMap[d.tname]["PrintInputParam"](d, NULL, NULL);
+  }
+  MLPACK_COUT_STREAM << ") {" << endl;
+  MLPACK_COUT_STREAM << endl;
+
+  // Restore CLI settings.
+  MLPACK_COUT_STREAM << "  CLI_RestoreSettings(\"" << CLI::ProgramName()
+      << "\")" << endl;
+  MLPACK_COUT_STREAM << endl;
+
+  // Handle each input argument's processing before calling mlpackMain().
```

Comment:
Before we start handling each of the input parameters and passing them to mlpack's `CLI` object, do we need to check that they are of the right type?  What would happen if, e.g., I passed `TRUE` to a matrix parameter?

Reply:
Matrix type parameter will be handled properly by [`to_matrix()`](https://github.com/Yashwants19/RcppMLPACK/blob/e9c825a382d47afcc707d6bee8a37d0e145085c1/R/matrix_utils.R#L19). All thanks to `Rcpp`, It is  handling "type checking" pretty nicely. 
For Example:
```R
> cf(algorithm = 1)
Error in CLI_SetParamString("algorithm", algorithm) : 
  Expecting a single string value: [type=double; extent=1].
```
But If you want I can implement a layer of type checking above this.

Reply:
Oh, nice!  This looks great to me.